### ISSUE TITLE
UX for extension disabled due to api version incompatibility

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#218646]
Discussed in the copilot sync today- we think that users may be confused when Copilot Chat gets disabled due to an api version incompatibility, because the chat icon disappears, and the only way to find out what happened is to go search for the extension in the viewlet.

I think that the ideal experience would be for me to be able to still register the chat viewlet and show some welcome view content in it, telling users that they need to update the extension. Do do this I need to be able to find out that the extension wasn't loaded due to the version mismatch- is there any way I can get this information, or can it be exposed somehow?

**Comment by sandy081** [comment]
You can get the installed extensions and look for the extension that contributes chat and if it has any validation errors.

Use following API to get installed extensions

https://github.com/microsoft/vscode/blob/ecc2d1ec7a3c6da3c305d8f8968a32e923f4f0e5/src/vs/workbench/contrib/extensions/common/extensions.ts#L121

You can get the installed extension model using `local` property on the `IExtension` 

https://github.com/microsoft/vscode/blob/ecc2d1ec7a3c6da3c305d8f8968a32e923f4f0e5/src/vs/workbench/contrib/extensions/common/extensions.ts#L97

**Comment by roblourens** [comment]
This works perfectly, added a notification specifically for Copilot Chat when it has validation errors, thanks

**Comment by roblourens** [comment]
@sandy081, can you check my PR? I'm a little confused. Both vscode and copilot chat have an api version update in the latest today. When I updated VS Code today, I saw my notification in every window

![Image](https://github.com/user-attachments/assets/dda1c545-ad8b-4877-8a46-16a9bbf88235)

but the extension is up to date

![Image](https://github.com/user-attachments/assets/410bfa29-331a-4d89-a37a-a6fc8a6bf756)

and when reloading the window, I don't see the notification again. I think you download the extension ahead of time, but the extension update would have been installed only after the window loads, right? So is there a moment when the old one is still loaded and validated?

**Comment by sandy081** [comment]
Here is the sequence of things explaining why this has happened:

- You bumped API version and Adopted it in Copilot chat extension day before (say yesterday)
- Today, You received VS Code Insider update and Updated it and Restarted VS Code Insider (Remember extension is not yet updated)
- VS Code Insider with bumped API version is opened with older version of chat extension. So extension is invalid and won't be activated. Your prompt check code gets kicked on this version of extension and prompt is shown.
- Parallely Extensions auto update starts and VS Code Insider updates the Chat extension to latest version
- Since the latest Chat extension version is compatible with the API, it is loaded and activated into Extension Host. 

In short, your prompt is shown during the time frame VS Code checks for updates and installs updates. 

Instead of the prompt, I would suggest to show an Error badge on Extensions View Container Icon to grasp user's attention. User can then go to the extension view and see what is happening.

**Comment by sandy081** [comment]
To Continue...

We have a limitation of auto updating the extension to the latest version even before updating the VS Code. Because, we do not know the API versions upfront.

**Comment by roblourens** [comment]
Yeah, that makes sense. I still worry that won't be obvious enough, though. Just really don't want to answer lots of issues from people who had chat suddenly disappear.

> Parallely Extensions auto update starts and VS Code Insider updates the Chat extension to latest version

Is there any way for me to know when this process is done?

Another idea I just had, maybe I can register a welcome view from vscode which would be cleared once copilot chat is able to load and register its participants?

**Comment by roblourens** [comment]
I kind of like the welcome view idea, I think that is possible to do with what exists today...

**Comment by roblourens** [comment]
That would be nice, I don't think we have the ability to have that kind of badge right now though. Sent a PR for the welcome view, lmk what you think.

**Comment by sandy081** [comment]
Extracted extensions UX part here - https://github.com/microsoft/vscode/issues/227047

**Comment by andreamah** [comment]
Are there some simple repro steps for this?

**Comment by roblourens** [comment]
- Install the Insiders for `36e4ddb4dbdf5d41bbeedfce2db4834085de2b87` from last week
- Install the Copilot Chat extension- you'll have to right click, "install specific version", pick the latest one in the list. There's a bug with the main "install" button that is fixed after that build
- Quit Insiders, open the latest Insiders again
- You should see the chat view with the welcome text and button
- Click the button, update Copilot Chat
- Now the chat view should load properly
